crates/sui-types/src/event.rs

```diff
@@ -73,33 +75,110 @@ pub enum TransferType {
     EnumDiscriminants,
     EnumVariantNames,
 )]
-#[strum_discriminants(name(EventType))]
+#[strum_discriminants(name(EventType), derive(Serialize, Deserialize, JsonSchema))]
 // Developer note: PLEASE only append new entries, do not modify existing entries (binary compat)
 pub enum Event {
     /// Move-specific event
-    MoveEvent(MoveObject),
+    MoveEvent {
+        package_id: ObjectID,
+        transaction_module: Identifier,
+        transaction_function: Identifier,
+        instigator: SuiAddress,
+        type_: StructTag,
+        #[serde_as(as = "Bytes")]
+        contents: Vec<u8>,
+    },
     /// Module published
-    Publish { package_id: ObjectID },
+    Publish {
+        instigator: SuiAddress,
+        package_id: ObjectID,
+    },
     /// Transfer objects to new address / wrap in another object / coin
     TransferObject {
+        package_id: ObjectID,
+        transaction_module: Identifier,
+        transaction_function: Identifier,
+        instigator: SuiAddress,
+        recipient: Owner,
         object_id: ObjectID,
         version: SequenceNumber,
         destination_addr: SuiAddress,
         type_: TransferType,
     },
     /// Delete object
-    DeleteObject(ObjectID),
+    DeleteObject {
+        package_id: ObjectID,
```

Comment:
I think we can keep both for now, there are talk in another thread about adding restriction in move to restrict EventType to be originated from the same package and module, we can remove package/module when that happened.

For now should I just leave it as it is? We can remove them later for space saving/optimisation later.

Reply:
Per conversation from yesterday it sounds like we now want to get rid of the "function" for clarity?

Reply:
PS2: I just realize that the thing we return to client `SuiEvent` is not authenticatable. To make it so, we need to at least return a TransactionDigest so clients can verify by pulling the txn by themselves. 

I will add that based on this PR .

Reply:
@longbowlu @patrickkuo ok, there are some differences between the EventStore and subscription module that we should discuss, related to this:
1. In EventStore, we store package_id and module as separate columns, to make the querying more flexible (let's say user queries only by package ID).   I think it's good to leave them as separate for flexibility.   In the future they would also compress better if left separate.
2. From an event subscription perspective it makes very little difference if package_id and module are in EventEnvelope or Event, they are 1:1.  However, it makes a big difference if we include them in Event for TransactionEffects, because including them means that the package id and module are repeated in each event, and that bloats the `TransactionEffects`.   

We can leave them here for now, but there are more compact ways to represent, like if each event had a reference to the source TransactionData number within the Transaction batch, and through that we can get the source MOVE package ID and module.

Reply:
@velvia  I re-read your to comment and have this q: " events all appear in TransactionEffects in a Vec, and we do not know which source SingleTransaction produced the event." A `TransactinoEffects` is only associated with a `Transaction` so there shouldn't be difficulties in linking back `Events` back to `Transaction` from that perspective.

after some pondering, I think I'm sold that we should put `package/module` into `EventEnvelope` instead of `Event`, reasons being:
1. the data bloat problem @velvia  pointed out above
2. most Events enums have `package` and `function`, so it's a good idea to put it into a wrapper struct ?
3. currently we seem to the `Event` enums, (except`packge/function`), they depict "what happened", but not "how/when/where it happened". For example we don't put timestamp (where) in `Event`. Following this logic, `sender` probably does not belong to `Event` either, except `TransferObject`.

I'm OK with leaving this PR as-is but let's sync on what to put in each `Event` enum next week

Reply:
Yes, `sender` should belong in the envelope as well since there is only one Sender per TX.  

Not sure what is meant about a wrapper struct